infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/LogReplicationClientConfigListener.java

```diff
@@ -0,0 +1,189 @@
+            if (entry.getOperation().equals(CorfuStreamEntry.OperationType.UPDATE)) {
+                String clientName = ((ClientRegistrationId) entry.getKey()).getClientName();
+                ReplicationModel model = ((ClientRegistrationInfo) entry.getPayload()).getModel();
+                LogReplication.ReplicationSubscriber subscriber = LogReplication.ReplicationSubscriber.newBuilder()
+                        .setClientName(DEFAULT_LOGICAL_GROUP_CLIENT).setModel(model).build();
+                // TODO (V2): Currently we add a default subscriber for logical group use case instead of listening on client
+                //  registration. Subscriber should be added upon registration after grpc stream for session creation is added.
+//                configManager.onNewClientRegister(subscriber);
+//                sessionManager.createOutgoingSessionsBySubscriber(subscriber);
+            } else {
+                // TODO (V2 / Chris/Shreay): add unregister API for clients
+            }
+        }
+    }
+
+    /**
+     * Handle client metadata table entries.
+     * @param clientMetadataTableEntries list of client metadata table entries
+     */
+    private void handleClientMetadataTableEntries(List<CorfuStreamEntry> clientMetadataTableEntries) {
+        Set<LogReplicationSession> impactedSessions = new HashSet<>();
+        log.info("clientMetadataTableEntries: {}", clientMetadataTableEntries);
+        if (clientMetadataTableEntries == null) {
+            log.warn("No client metadata table entries found!");
+            return;
+        }
+
+        for (CorfuStreamEntry entry : clientMetadataTableEntries) {
+            ClientDestinationInfoKey clientInfo = (ClientDestinationInfoKey) entry.getKey();
+            LogReplication.ReplicationSubscriber subscriber = LogReplication.ReplicationSubscriber.newBuilder()
+                    .setClientName(DEFAULT_LOGICAL_GROUP_CLIENT).setModel(clientInfo.getModel()).build();
+            if (entry.getOperation().equals(CorfuStreamEntry.OperationType.UPDATE)) {
+                DestinationInfoVal sinksInfo = (DestinationInfoVal) entry.getPayload();
+                impactedSessions = configManager.onGroupDestinationsChange(subscriber, clientInfo.getGroupName(),
+                        sinksInfo.getDestinationIdsList());
+            } else if (entry.getOperation().equals(CorfuStreamEntry.OperationType.DELETE)) {
+                impactedSessions = configManager.onGroupDestinationsChange(subscriber, clientInfo.getGroupName(),
+                        new ArrayList<>());
+            }
+
+            if (impactedSessions != null) {
+                log.info("Sessions that a forced snapshot sync will be triggered: {}", impactedSessions);
+                impactedSessions.forEach(sessionManager::enforceSnapshotSync);
+            }
+        }
+    }
+
+    /**
+     * Unsubscribe this stream listener to stop
+     */
+    public void stop() {
```

Comment:
Did not find this getting invoked on leadership loss

Reply:
Thanks! I might have lost that part when doing the rebase, added it back